File: app/Lecture.php
<?php

namespace App;

use Illuminate\Database\Eloquent\Model;

class Lecture extends Model
{
    protected $fillable = [
        'subject', 'user_id', 'begin', 'end',
    ];

    public function user() {
      return $this->belongsTo('App\User');
    }

    public function entries() {
      return $this->hasMany('App\Entry');
    }
}

File: app/models/lecture.rb
# Model class for the +lectures+ table
class Lecture < ApplicationRecord
  scope :in_progress, -> { where("dtend > ?", Time.current) }
  belongs_to :user
  has_many :entries, dependent: :nullify
  validates :dtstart, presence: true
  validates :device_id, presence: true
end
